rpcs3/Emu/Cell/Modules/cellAvconfExt.cpp

```diff
@@ -368,12 +395,31 @@ error_code cellAudioInRegisterDevice(u64 deviceType, vm::cptr<char> name, vm::pt
 		return CELL_AUDIO_IN_ERROR_ILLEGAL_PARAMETER;
 	}
 
-	return not_an_error(0); // device number
+	auto& av_manager = g_fxo->get<avconf_manager>();
+	const std::lock_guard lock(av_manager.mutex);
+
+	std::optional<CellAudioInDeviceInfo> info = av_manager.get_device_info(name);
+	if (!info || !memchr(info->name, '\0', sizeof(info->name)))
+	{
+		// TODO
+		return CELL_AUDIO_IN_ERROR_DEVICE_NOT_FOUND;
+	}
+
+	auto& mic_thr = g_fxo->get<mic_thread>();
+	const std::lock_guard mic_lock(mic_thr.mutex);
+	const u32 device_number = mic_thr.register_device(info->name);
```

Comment:
From doc I think the game can repeatedly call this function, it'll just return the same device number again.
We're missing a check to make sure not to register the same device twice, either here or in mic_context::register_device.